### PR TITLE
Add Glean instrumentation to record in-memory-cache misses.

### PR DESCRIPTION
**Pull request by jhugman**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[ac: android-components-branch-name]` and/or `[fenix: fenix-branch-name]` to the PR title.

### CODE REVIEW

**Comment by jhugman** [comment]
# Request for data collection review form

1) What questions will you answer with this data?

The purpose of this event is to help with timing of onboarding experiments to understand how the timing of Nimbus experiment enrollment calculations can cause issues with how early the application requests feature configuration information. This error will answer how often this happens in the field, as well as give us ideas of how that may bias experiments.

1) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:

These metrics are important for monitoring of Nimbus experiments, to ensure proper experiment execution, especially for early in launch experiments and first run experiments.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

No existing telemetry or error provides this information.

4) Can current instrumentation answer these questions?

No.

An [existing event][1] records a similar event, but this and that differ in terms of:

1. [`feature_request_error`][1] records when the feature was requested **before** the SDK has **started** initialization.
2. This event records when the **before** the SDK has **finished** initialization.

[1]: https://github.com/mozilla/application-services/pull/5091

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.   

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>Race In Memory Cache Event</td>
    <td>Category 1 - Technical Data</td>
    <td>[Jira EXP-2743](https://mozilla-hub.atlassian.net/browse/EXP-2743)</td>
  </tr>
</table>

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.

This collection is documented in the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/) and in the metrics.yaml in /components/nimbus

7) How long will this data be collected?  Choose one of the following:

I want to permanently monitor this data - Travis Long (tlong@mozilla.com)

8) What populations will you measure?

All channels, countries, and locales

9) If this data collection is default on, what is the opt-out mechanism for users?

Through the standard telemetry preference in the application settings

10) Please provide a general description of how you will analyze this data.

Through Looker and Redash.

11) Where do you intend to share the results of your analysis?

With experiment team members 

12)  Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?

No

File: components/nimbus/metrics.yaml [hunk 1]
```diff
@@ -135,3 +135,21 @@ nimbus_health:
...
+        type: string
+        description: The feature id of the configuration that was requested
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/EXP-2743
+    data_reviews:
+      - https://github.com/mozilla/application-services/pull/5118#issuecomment-1235827006
+    data_sensitivity:
+      - technical
+    notification_emails:
+      - tlong@mozilla.com, telemetry-team@mozilla.com
```
Comment (travis79) [inline components/nimbus/metrics.yaml:154]:
```suggestion
      - tlong@mozilla.com
      - telemetry-team@mozilla.com
```

**Comment by travis79** [comment]
> # Request for data collection review form
> 
>     1. What questions will you answer with this data?
> 
> 
> The purpose of this event is to help with timing of onboarding experiments to understand how the timing of Nimbus experiment enrollment calculations can cause issues with how early the application requests feature configuration information. This error will answer how often this happens in the field, as well as give us ideas of how that may bias experiments.
> 
>     1. Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
> 
> 
> These metrics are important for monitoring of Nimbus experiments, to ensure proper experiment execution, especially for early in launch experiments and first run experiments.
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
> No existing telemetry or error provides this information.
> 
>     4. Can current instrumentation answer these questions?
> 
> 
> No.
> 
> An [existing event](https://github.com/mozilla/application-services/pull/5091) records a similar event, but this and that differ in terms of:
> 
>     1. [`feature_request_error`](https://github.com/mozilla/application-services/pull/5091) records when the feature was requested **before** the SDK has **started** initialization.
> 
>     2. This event records when the **before** the SDK has **finished** initialization.
> 
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.
> 
> 
> **Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Race In Memory Cache Event 	Category 1 - Technical Data 	[Jira EXP-2743](https://mozilla-hub.atlassian.net/browse/EXP-2743)
> 
>     6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
> This collection is documented in the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/) and in the metrics.yaml in /components/nimbus
> 
>     7. How long will this data be collected?  Choose one of the following:
> 
> 
> I want to permanently monitor this data - Travis Long ([tlong@mozilla.com](mailto:tlong@mozilla.com))
> 
>     8. What populations will you measure?
> 
> 
> All channels, countries, and locales
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
> Through the standard telemetry preference in the application settings
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
> Through Looker and Redash.
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
> With experiment team members
> 
>     12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
> 
> 
> No

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/).

2) Is there a control mechanism that allows the user to turn the data collection on and off?

Yes, through the "Send Usage Data" preference in the application settings.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

Travis Long <tlong@mozilla.com> will be responsible for monitoring the permanent collection of this data.

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 1, Technical data

5) Is the data collection request for default-on or default-off?

Default-on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No

7) Is the data collection covered by the existing Firefox privacy notice?

Yes

8) Does the data collection use a third-party collection tool?

No

## Result

data-review+

File: components/nimbus/metrics.yaml (hunk 1 above)
Comment (travis79) [reply components/nimbus/metrics.yaml:154]: Just this nit, everything else looks good to me.

**Review by travis79 — APPROVED** [review]

**Pull request merged**